File: MyLanguageDB/src/main/java/ru/herooo/mylanguagedb/entities/WordStatusHistory.java
package ru.herooo.mylanguagedb.entities;

import jakarta.persistence.*;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
public class WordStatusHistory {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "word_status_history_id_seq")
    @SequenceGenerator(name = "word_status_history_id_seq", sequenceName = "word_status_history_id_seq", allocationSize = 1)
    long id;

    @Column(name = "date_of_start")
    private LocalDateTime dateOfStart;

    @Column(name = "date_of_end")
    private LocalDateTime dateOfEnd;

    @ManyToOne
    @JoinColumn(name = "word_id")
    private Word word;

    @ManyToOne
    @JoinColumn(name = "word_status_id")
    private WordStatus wordStatus;

    public long getId() {
        return id;
    }

    public LocalDateTime getDateOfStart() {
        return dateOfStart;
    }

    public void setDateOfStart(LocalDateTime dateOfStart) {
        this.dateOfStart = dateOfStart;
    }

    public LocalDateTime getDateOfEnd() {
        return dateOfEnd;
    }

    public void setDateOfEnd(LocalDateTime dateOfEnd) {
        this.dateOfEnd = dateOfEnd;
    }

    public Word getWord() {
        return word;
    }

    public void setWord(Word word) {
        this.word = word;
    }

    public WordStatus getWordStatus() {
        return wordStatus;
    }

    public void setWordStatus(WordStatus wordStatus) {
        this.wordStatus = wordStatus;
    }
}

File: MyLanguageWeb/src/main/resources/static/js/dto/word_status_history.js
import {
    WordResponseDTO
} from "./word.js";

import {
    WordStatusResponseDTO
} from "./word_status.js";

export class WordStatusHistoryResponseDTO {
    id;
    dateOfStart;
    dateOfEnd;
    word;
    wordStatus;

    constructor(wordStatusHistoryJson) {
        if (wordStatusHistoryJson) {
            this.id = wordStatusHistoryJson["id"];
            this.dateOfStart = wordStatusHistoryJson["date_of_start"];
            this.dateOfEnd = wordStatusHistoryJson["date_of_end"];
            this.word = new WordResponseDTO(wordStatusHistoryJson["word"]);
            this.wordStatus = new WordStatusResponseDTO(wordStatusHistoryJson["word_status"]);
        }
    }
}